src/redux/reducers/dataReducer.js

```diff
@@ -25,24 +26,34 @@ export default function(state = initialState, action) {
         postList: action.payload,
         isLoading: false,
       };
+    case NEW_POST:
+      return {
+        ...state,
+        postList: [action.payload, ...state.postList],
```

Comment:
The order here matters. Whatever is last is kept. if state.postList has the same content as action.payload, it will overwrite that payload.

Reply:
The payload being returned isn't a postList, but a single post being added to the beginning of the array.

Reply:
Oh right... misread that.